packages/valory/skills/abstract_round_abci/models.py

```diff
@@ -329,11 +329,14 @@ def _ensure_setup(
         enforce(bool(self.setup_params), "`setup` params contain no values!")
 
         for key, type_ in necessary_params.items():
-            value = self.setup_params.get(key, None)
-            if value is None:
+            # check that the key is present, note that None is acceptable for optional keys
+            missing_value = object()
+            value = self.setup_params.get(key, missing_value)
+            if value is missing_value:
```

Comment:
Maybe we can import the `VALUE_NOT_PROVIDED` instead:

```suggestion
            value = self.setup_params.get(key, VALUE_NOT_PROVIDED)
            if value is VALUE_NOT_PROVIDED:
```